### PR TITLE
LPD-24932 Some Solutions Flow Adjusts and ZodSchema validations

### PR DESCRIPTION
**Pull request by welitonpe**

### CODE REVIEW

**Comment by welitonpe** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/PublishSolutionOutlet.tsx [hunk 1]
```diff
@@ -123,14 +128,14 @@ const PublishSolutionOutlet = () => {
 							displayType="secondary"
 							onClick={() => onSaveAsDraft().then(onExit)}
 						>
-							Save as a draft & exit
+							{i18n.translate('save-as-a-draft-&-exit')}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/PublishSolutionOutlet.tsx:131]:
```suggestion
							{i18n.translate('save-as-a-draft-exit')}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/i18n/en_US.ts [hunk 2]
```diff
@@ -275,6 +281,7 @@ export default {
 	'review-and-submit-solution': 'Review and submit solution',
 	'review-the-new-publisher-info-and-the-liferay-marketplace-terms-before-proceeding':
 		'Review the new publisher info and the Liferay Marketplace terms before proceeding',
+	'save-as-a-draft-&-exit': 'Save as a draft & exit',
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/i18n/en_US.ts:284]:
```suggestion
	'save-as-a-draft-exit': 'Save as a draft & exit',
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/i18n/en_US.ts [hunk 3]
```diff
@@ -162,6 +165,8 @@ export default {
 		'If the app is compatible with different updates of 7.4, please upload multiple packages for each update or update compatibility range.',
 	'if-you-are-requesting-the-account-on-behalf-of-a-business-,-please-note-the-business-name':
 		'If you are requesting the account on behalf of a business, please note the business name.',
+	'important-Images-will-be-displayed-following-thenumerical-order-above':
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/i18n/en_US.ts:168]:
```suggestion
	'important-images-will-be-displayed-following-the-numerical-order-above':
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/i18n/en_US.ts [hunk 4]
```diff
@@ -162,6 +165,8 @@ export default {
...
+		'Important: Images will be displayed following thenumerical order above',
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/i18n/en_US.ts:169]:
```suggestion
		'Important: Images will be displayed following the numerical order above',
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 5]
```diff
@@ -157,6 +160,7 @@ export type SolutionInitialState = {
 	references: {
 		vocabulariesAndCategories: any;
 	};
+	submit: boolean;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:163]:
```suggestion
	termsAndConditions: boolean;
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 6]
```diff
@@ -189,6 +193,7 @@ const solutionInitialState: SolutionInitialState = {
 		tags: [],
 	},
 	references: {vocabulariesAndCategories: {}},
+	submit: false,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:196]:
```suggestion
	termsAndConditions: false,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx [hunk 7]
```diff
@@ -3,8 +3,277 @@
...
+							symbol="phone"
+						/>
+					</ContentReview.Paragraph>
+				</ContentReview.Section>
+
+				<ContentReview.Separator />
+
+				<ContentReview.Section>
+					<ContentReview.Header as="h2" path="../contact">
+						<h2 className="mb-0">{i18n.translate('contact-us')}</h2>
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx:247]:
```suggestion
{i18n.translate('contact-us')}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx [hunk 8]
```diff
@@ -3,8 +3,277 @@
...
+						href={contactUs}
+						linkLabel={i18n.translate('email')}
+						symbol="envelope-closed"
+					/>
+				</ContentReview.Section>
+			</ContentReview>
+
+			<div className="d-flex my-5">
+				<ClayCheckbox
+					checked={!!context.submit}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx:259]:
```suggestion
					checked={context.submit}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx [hunk 9]
```diff
@@ -3,8 +3,277 @@
...
+						<p
+							dangerouslySetInnerHTML={{
+								__html: DOMPurify.sanitize(company.description),
+							}}
+						/>
+					</ContentReview.Paragraph>
+					<ContentReview.Paragraph>
+						<ContentReview.SuportLink
+							href={company.website}
+							linkLabel={i18n.translate('publisher-website-url ')}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx:225]:
```suggestion
							linkLabel={i18n.translate('publisher-website-url')}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx [hunk 10]
```diff
@@ -3,8 +3,277 @@
...
+					</ContentReview.Section>
+				)}
+
+				<ContentReview.Separator />
+
+				<ContentReview.Section>
+					<ContentReview.Header as="h2" path="../company">
+						<h2 className="mb-0">
+							{i18n.translate('company-profile')}
+						</h2>
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx:210]:
```suggestion
							{i18n.translate('company-profile')}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx [hunk 11]
```diff
@@ -3,8 +3,277 @@
...
+
+				<ContentReview.Separator />
+
+				{!!details.length && (
+					<ContentReview.Section>
+						<ContentReview.Header as="h2" path="../details">
+							{i18n.translate('solution-details')}
+						</ContentReview.Header>
+						{details.map((block, index) => {
+							return (
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx:149]: return direct

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx [hunk 12]
```diff
@@ -3,8 +3,277 @@
...
+				<h2 className="mb-0 mr-3">
+					{i18n.translate('solution-submission')}
+				</h2>
+				<Tooltip
+					tooltip={i18n.translate('more-info')}
+					tooltipText={i18n.translate('more-info')}
+				/>
+			</span>
+
+			<hr className="mb-5 mt-2" />
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Submit.tsx:38]:
```suggestion
			<ContentReview.Separator />
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/components/ContentReview/ContentReviewHeader.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import classNames from 'classnames';
+import {ReactNode} from 'react';
+import {Link} from 'react-router-dom';
+
+interface ContentReviewHeaderProps {
+	as?: 'span' | 'h1' | 'h2' | 'h3';
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/components/ContentReview/ContentReviewHeader.tsx:11]:
```suggestion
	as?: 'h1' | 'h2' | 'h3' | 'span';
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/components/ContentReview/ContentReviewHeader.tsx [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+			className={classNames(
+				'align-items-center d-flex flex-row justify-content-between mb-5',
+				className
+			)}
+		>
+			<Wrapper>{children}</Wrapper>
+
+			<Link to={path}>
+				<strong>Edit</strong>
+			</Link>
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/components/ContentReview/ContentReviewHeader.tsx:36]:
```suggestion
			<Link className="font-weight-bold" to={path}>{i18n.translate("edit")}</Link>
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/components/ContentReview/index.tsx [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+	children,
+}) => <div className="border p-5 rounded-lg">{children}</div>;
+
+ContentReview.Block = ContentReviewBlock;
+ContentReview.Header = ContentReviewHeader;
+ContentReview.ImageInfo = ContentReviewImageInfo;
+ContentReview.Paragraph = ContentReviewParagraph;
+ContentReview.Section = ContentReviewSection;
+ContentReview.Separator = ContentReviewSeparator;
+ContentReview.SuportLink = ContentReviewSuportLink;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/components/ContentReview/index.tsx:42]:
```suggestion
ContentReview.SupportLink = ContentReviewSupportLink;
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 16]
```diff
@@ -89,7 +89,7 @@ export enum SolutionTypes {
 	SET_PRODUCT_ID = 'SET_PRODUCT_ID',
 	SET_PROFILE = 'SET_PROFILE',
 	SET_UPDATE_BLOCK = 'SET_UPDATE_BLOCK',
-	SET_SUBMIT = 'SET_SUBMIT',
+	SET_TERMS_AND_CONDITIONS = 'SET_TERMS_AND_CONDITIONS',
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:92]: Before SET_UPDATE_BLOCK

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Pull request closed without merging**